c/models/py_ftrl.cc

```diff
@@ -331,9 +331,14 @@ oobj Ftrl::fit(const PKArgs& args) {
                          << "one column";
     }
 
-    if (dt_y_val->get_ocolumn(0).stype() != dt_y->get_ocolumn(0).stype()) {
-      throw ValueError() << "Validation target frame must have the same "
-                            "stype as the target frame";
+
+    SType stype = dt_y->get_ocolumn(0).stype();
+    SType stype_val = dt_y_val->get_ocolumn(0).stype();
+
+    if (info(stype).ltype() != info(stype_val).ltype()) {
+      throw TypeError() << "Training and validation target columns must have "
+                        << "the same ltype, got: `" << info(stype).ltype_name()
```

Comment:
it's probably better to add an `Exception::operator<<(LType)`

Reply:
I guess you mean `Error::operator<<(LType)`? Added now.